monkey/monkey_island/cc/ui/src/components/ui-components/Select.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import {InputLabel, FormControl, Select, MenuItem, SelectChangeEvent} from '@mui/material';
+
+
+type Option = {
+  label: string,
+  value: string,
+}
+
+type SelectProps = {
+  placeholder: string,
+  options: Option[],
+  selectedOption: Option,
+  onChange: (event: SelectChangeEvent) => void
+}
+
+const SelectComponent = (props: SelectProps) => {
```

Comment:
Why is it outside of the return statement?

Reply:
WDYM?

Reply:
there's no need to create a variable. just do it inside the jsx

Reply:
Destruct the props e.g.:  `const {options} = {...props};`.
Then you can just use `options`.

What about enabling an 'All' option?

Reply:
> there's no need to create a variable. just do it inside the jsx
What variable? Are you sure you commented on the right lines? Or are you talking about `selectOptions` var?

> What about enabling an 'All' option?
What exactly is 'All'?

Reply:
Yes, the `selectOptions`.
'All' - for all options to be selected (e.g.: in TypeFilter it will be all the types)